snap/hooks/configure

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh -e
+
+set -eux
+
+# Note: the future we might want to handle scenarios where we reseting the URI restarts the daemon
+
+# Create the config file that is used by pbm
+CONFIG_FILE="$SNAP_COMMON/pbm_config.yaml"
+if [ ! -f "${CONFIG_FILE}" ]; then
+    echo "Using default config ${CONFIG_FILE}"
+    touch "$CONFIG_FILE"
+fi
+
+# set the config file for pbm
+if [ -n "$(snapctl get uri)" ]
+then
+    export PBM_MONGODB_URI="$(snapctl get uri)"
+    # pbm requires that a config file was set up in order to facilitate further congurations. Note 
+    # snaps require special permissions to read files outside of the typical snap directories so
+    # it is important that we set this to a file the snap is able to access.
+    # Further, this command requires that the uri has already been set so we must do it after the
+    # user sets the uri
+    "$SNAP/bin/pbm" config --file "$CONFIG_FILE"
+fi
+
+# the pbm-agent daemon is not run as root meaning it does not have access write to SNAP_DATA, so
+# we need to create a child dir and give it access.
+if [ ! -d "$SNAP_DATA/pbm-agent-dir" ]; then
+    mkdir "$SNAP_DATA/pbm-agent-dir"
+    chmod 770 "$SNAP_DATA/pbm-agent-dir"
+    chown snap_daemon "$SNAP_DATA/pbm-agent-dir"
```

Comment:
Do we need IF condition here? I think it is enough:
```
    mkdir -p "${SNAP_DATA}/pbm-agent-dir"
    chmod 770 "${SNAP_DATA}/pbm-agent-dir"
```
Both commands are fast, no need to optimize performance here.
Also the line 44 will overwrite this `chown` anyway:

```
chown -R snap_daemon:root $SNAP_DATA
```

I believe both IF conditions in lines 28 and 36 will be executed the first snap start only.